mcda/configuration/config.py

```diff
@@ -8,19 +8,33 @@
 
 
 # noinspection PyMethodMayBeStatic
-class Config():
+class Config:
     """
-    Class Configuration
-
-    keys expected in the input dictionary are
-    input_matrix_path: path to the input matrix
-    marginal_distribution_for_each indicator: list of marginal distributions, one for each indicator
-    polarity_for_each_indicator: list of polarities, one for each indicator
-    monte_carlo_runs: number of MC runs
-    num_cores: number of cores used in the parallelization
-    monte_carlo_runs: list of weights, one for each indicator
-    output_path: path to the output file
-
+    Class representing configuration settings.
+
+    This class encapsulates the configuration settings.
+    It expects the following keys in the input dictionary:
+    - input_matrix_path: path to the input matrix file.
+    - polarity_for_each_indicator: list of polarities, one for each indicator.
+    - sensitivity: sensitivity configuration.
+    - robustness: robustness configuration.
+    - monte_carlo_sampling: Monte Carlo sampling configuration.
+    - output_path: path to the output file.
+
+    Attributes:
+    _valid_keys (List[str]): list of valid keys expected in the input dictionary.
+    _list_values (List[str]): list of keys corresponding to list values.
+    _str_values (List[str]): list of keys corresponding to string values.
+    _int_values (List[str]): list of keys corresponding to integer values.
+    _dict_values (List[str]): list of keys corresponding to dictionary values.
+    _keys_of_dict_values (Dict[str, List[str]]): dictionary containing keys and their corresponding sub-keys.
+
+    Methods:
+    __init__(input_config: dict): instantiate a configuration object.
+    _validate(input_config, valid_keys, str_values, int_values, list_values, dict_values): validate the input configuration.
```

Comment:
I think this line will give linting error. Seems to be crossing 120. 
Did you run linting on this branch?